x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/notifications/MockTransformAuditor.java

```diff
@@ -134,28 +147,52 @@ public boolean innerMatch(final Level level, final String resourceId, final Stri
     public static class SeenAuditExpectation extends AbstractAuditExpectation {
 
         public SeenAuditExpectation(String expectedName, Level expectedLevel, String expectedResourceId, String expectedMessage) {
-            super(expectedName, expectedLevel, expectedResourceId, expectedMessage);
+            super(expectedName, expectedLevel, expectedResourceId, expectedMessage, 1);
         }
 
         @Override
         public void assertMatched() {
-            assertThat("expected to see " + expectedName + " but did not", saw, equalTo(true));
+            assertThat("expected to see " + expectedName + " but did not", count, equalTo(expectedCount));
```

Comment:
If would be good to distinguish the case where we don't see the message at all to the case where we see it too many times.  Otherwise, if we ever get this test failure we won't know which scenario was to blame.

In fact, just reading further the error message in `MultipleSeenAuditExpectation` is ideal, so maybe just delete this class and always use `MultipleSeenAuditExpectation` with expected count 1.

Reply:
I realize that I changed the way it worked, before it just flipped a bool, so it was "seen at least once", with the count I changed it to "exactly once". However the tests pass and "exactly once" is probably what we want anyway.

I will merge the 2 classes and keep the constructor that defaults to "seen once" as convenience. That way it does not trigger further refactorings.